### DISCUSSION TITLE
Field in BaseModel can accept a list of alias

### DISCUSSION THREAD

**Discussion by belal-b-ali** [pydantic/pydantic#2964] [Feature Request]
- So I'm currently working on DataMigration project, I'm using `BaseModel` to convert the data, and validate it, the problem is I'm migrating from two different systems and each system has its own field name.

- For example, the one system used `fname` and the other one used `first` and the system I'm working on is first_name, I'm looking to add something like this `first_name: str = Field(..., alias=['first', 'fname'])`. Thanks in advance!

**Comment by lesnik512** [comment]
Looks like it was already discussed [here](https://github.com/samuelcolvin/pydantic/issues/324#issuecomment-448579228)

And by the link there is some solution posted. May it will help

**Reply by belal-b-ali** [reply]
Yup, I saw it, the thing is that I have more than two names so using `allow_population_by_field_name` is useless. Thanks in advance!

**Comment by lesnik512** [comment]
What about sth like this?

```python
from pydantic import BaseModel, root_validator

class SomeModel(BaseModel):
    first_name: str

    @root_validator(pre=True)
    def parse_field(cls, values):  # pylint: disable=no-self-argument
        aliases = ('first', 'fname')
        while True:
            if "first_name" in values:
                break
            for x in aliases:
                if x in values:
                    values["first_name"] = values[x]
                    break
            break
        return values
        
SomeModel(first="some_name"), SomeModel(fname="some_name2")
```

**Reply by belal-b-ali** [reply]
oh, this will be helpful thanks alot!

**Reply by lesnik512** [reply]
I made a mistake with infinite loop. I fixed it by `break` at the end of it
